src/GraphQL.Tests/Subscription/ShouldlyExtensions.cs

```diff
@@ -57,4 +58,16 @@ public static ExecutionResult ShouldBeSuccessful(this ExecutionResult result)
         result.Executed.ShouldBeTrue();
         return result;
     }
+
+    public static void ShouldBeWhen(this object? obj, bool condition, object onTrue, object onFalse) =>
+        obj.ShouldBe(condition ? onTrue : onFalse);
+
+    public static void ShouldThrowWhen<TException>(bool condition, Action action)
```

Comment:
I wouldn't really do this one either, personally, mainly because it can't be an extension method.  So we can't use `Should.Thow` like we can anywhere else.  But ok if you want to for your tests.

I would comment that typically within GraphQL.NET we examine the message from the exception to verify that it is the one intended.  I'm not firm on this (just stating what the existing code is like), but you could modify `void` to be `TException?` along with an annotation that indicates it's not-null when `condition` is `true`.  Then it could be used like this:

```cs
ShouldThrowWhen(condition, () => DoStuff())
    ?.Message.ShouldBe("abcd");
```

Personally I think verifying the exception type is often suitable, but sometimes verifying the message is important when the exception type is rather generic.